src/utils.ts

```diff
@@ -380,12 +380,22 @@ export async function getOwnerSafes(owner: string, config: Config): Promise<stri
 
 // Get token balances for an address.
 export async function getTokens(address: string, config: Config):
-  Promise<Array<{ tokenName: string; tokenLogo: string }>>
+  Promise<Array<{ name: string; symbol: string; decimals: number; logo: string; balance: BigNumber }>>
 {
-  await config.provider.send("alchemy_getTokenBalances", [address, config.tokens]);
+  const userBalances = await config.provider.send("alchemy_getTokenBalances", [address, "DEFAULT_TOKENS"]);
+  const balances = userBalances.tokenBalances.filter((token: any) => {
+    // alchemy_getTokenBalances sometimes returns 0x and this does not work well with ethers.BigNumber
+    if (token.tokenBalance !== "0x") {
+      if (!BigNumber.from(token.tokenBalance).isZero()) {
+        return token;
+      }
+    }
+  }).map(async (token: any) => {
+    const tokenMetaData = await config.provider.send("alchemy_getTokenMetadata", [token.contractAddress]);
+    return { ...tokenMetaData, balance: BigNumber.from(token.tokenBalance) };
+  });
```

Comment:
Does this fail gracefully if `alchemy_getTokenMetadata` is not supported by the endpoint? ie. just return an empty list.

Reply:
Yes, if `balances` is just an empty array, the condition `{#if tokens}` in the Orgs View component avoids rendering anything.
And if the endpoint does not support `alchemy_getTokenMetadata` the Promise get's rejected and it gets catched in the `getOrgsTreasury` function and no balance is being shown.